cobalt/xhr/xml_http_request.cc

```diff
@@ -518,11 +518,12 @@ void XMLHttpRequestImpl::Send(
     const base::Optional<XMLHttpRequest::RequestBodyType>& request_body,
     script::ExceptionState* exception_state) {
   error_ = false;
-  bool in_service_worker = environment_settings()
-                               ->context()
-                               ->GetWindowOrWorkerGlobalScope()
-                               ->IsServiceWorker();
-  if (!in_service_worker && method_ == net::URLFetcher::GET) {
+  auto* context = environment_settings()->context();
+  bool in_service_worker =
+      context->GetWindowOrWorkerGlobalScope()->IsServiceWorker();
+  bool has_active_service_worker =
+      !in_service_worker && context->active_service_worker();
+  if (has_active_service_worker && method_ == net::URLFetcher::GET) {
```

Comment:
If I understand it correctly, the change here is making the bug happen less frequently, like when there is no active service worker, but it doesn't resolve the bug for SW implementation.
The code change here looks good to me. But based on what @jellefoks called out in the previous PR,  https://github.com/youtube/cobalt/pull/1499#discussion_r1320391000, can we do a quick fix for that as well?